stdlib/public/Concurrency/TaskSleepDuration.swift

```diff
@@ -21,33 +22,24 @@ extension Task where Success == Never, Failure == Never {
     tolerance: Duration?,
     clock: _ClockID
   ) async throws {
-    // Allocate storage for the storage word.
-    let wordPtr = UnsafeMutablePointer<Builtin.Word>.allocate(capacity: 1)
-
-    // Initialize the flag word to "not started", which means the continuation
-    // has neither been created nor completed.
-    Builtin.atomicstore_seqcst_Word(
-        wordPtr._rawValue, SleepState.notStarted.word._builtinWordValue)
+    // Create a token which will initially have the value "not started", which
+    // means the continuation has neither been created nor completed.
+    let token = Atomic<SleepState>(.notStarted)
```

Comment:
If this actually works lifetime wise then it's awesome to get rid of the allocation!

Reply:
As it turns out, I don't think this prototype gets rid of a heap allocation at all - it just obscures it. I'm pretty sure that because `token` is captured by an escaping closure, its storage becomes a heap allocated closure context that needs to live as long as instances of the closure. So I think this not only doesn't remove a heap allocation but it actually also introduces additional refcounting traffic that isn't present in the wrapper based implementation.

This implementation may still be worth it from a simplicity perspective, though.